js/main.js

```diff
@@ -0,0 +1,113 @@
+    $h2ProgrammingPunchline.appendChild(progPunchline);
+  });
+  xhr.send();
+}
+
+function generalJokes() {
+  var xhr = new XMLHttpRequest();
+  xhr.open('GET', 'https://official-joke-api.appspot.com/random_joke');
+  xhr.responseType = 'json';
+  xhr.addEventListener('load', function () {
+    var strJokeGeneral = xhr.response.setup;
+    var genJoke = document.createTextNode(strJokeGeneral);
+    $h2GeneralSetup.appendChild(genJoke);
+
+    var strPunchlineGeneral = xhr.response.punchline;
+    var genPunchline = document.createTextNode(strPunchlineGeneral);
+    $h2GeneralPunchline.appendChild(genPunchline);
+  });
+  xhr.send();
+}
+
+var $homePage = document.querySelector('#home-page');
+var $generalSetup = document.querySelector('.general-jokes-setup');
+var $generalPunchline = document.querySelector('.general-jokes-punchline');
+var $programmingSetup = document.querySelector('.programming-jokes-setup');
+var $programmingPunchline = document.querySelector('.programming-jokes-punchline');
+
+var $chooseGeneral = document.querySelector('.general-jokes-button');
+var $chooseProgramming = document.querySelector('.programming-jokes-button');
+var $tellMeGeneral = document.querySelector('.tell-me-general');
+var $tellMeProgramming = document.querySelector('.tell-me-programming');
+var $anotherGeneral = document.querySelector('.another-general');
+var $anotherProgramming = document.querySelector('.another-programming');
+
+function handleClick(event) {
+  if (event.target === $chooseGeneral) {
+    $homePage.classList.add('hidden');
+    $generalSetup.classList.remove('hidden');
+    generalJokes();
+  }
+  if (event.target === $chooseProgramming) {
+    $homePage.classList.add('hidden');
+    $programmingSetup.classList.remove('hidden');
+    programmingJokes();
+  }
+  if (event.target === $tellMeGeneral) {
+    $generalSetup.classList.add('hidden');
+    $generalPunchline.classList.remove('hidden');
+    $h2GeneralSetup.textContent = '';
+  }
+  if (event.target === $tellMeProgramming) {
+    $programmingSetup.classList.add('hidden');
+    $programmingPunchline.classList.remove('hidden');
+    $h2ProgrammingSetup.textContent = '';
+  }
+  if (event.target === $anotherGeneral) {
+    $generalPunchline.classList.add('hidden');
+    $generalSetup.classList.remove('hidden');
+    generalJokes();
+    $h2GeneralPunchline.textContent = '';
+  }
+  if (event.target === $anotherProgramming) {
+    $programmingPunchline.classList.add('hidden');
+    $programmingSetup.classList.remove('hidden');
+    programmingJokes();
+    $h2ProgrammingPunchline.textContent = '';
+  }
+}
+
+$chooseGeneral.addEventListener('click', handleClick);
+$chooseProgramming.addEventListener('click', handleClick);
+$tellMeGeneral.addEventListener('click', handleClick);
+$tellMeProgramming.addEventListener('click', handleClick);
+$anotherGeneral.addEventListener('click', handleClick);
+$anotherProgramming.addEventListener('click', handleClick);
```

Comment:
Instead of having an individual event listener for each element, you should put one class on each of these elements, and apply that event listener to all elements with that class.